tools/isobuild/bundler.js

```diff
@@ -792,6 +806,33 @@ class Target {
                 this.nodeModulesDirectories[unibuild.nodeModulesPath] = nmd;
               }
               f.nodeModulesDirectory = nmd;
+
+              if (!archinfo.matches(this.arch, unibuild.arch)) {
+                // The unibuild we're trying to include doesn't work for the
+                // bundle target (eg, os.osx.x86_64 instead of os.linux.x86_64)!
+                // Hopefully this is because we specially enabled the feature
+                // that leads to this.
+                if (!this.providePackageJSONForUnavailableBinaryDeps) {
+                  throw Error("mismatched arch without special feature enabled "
```

Comment:
Huh, really? There are places in tool where we use "throw Error(string)"
(not some specific subclass that can be detected, just a random string) for
user visible issues? How do we manage to avoid showing stack traces for
them without also hiding stack traces for coding errors?